packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -21,83 +18,181 @@ export interface Props extends Omit<CardContainerProps, 'disableEvents' | 'disab
+        }
 
-      if (found && React.isValidElement(found)) {
-        return React.cloneElement(found, { disabled, styles, ...found.props });
-      }
-      return found;
-    }
+        if (validComponents.includes(componentType)) {
+          if (componentType === 'Heading') {
+            return React.cloneElement(c, { disabled, styles, href, onClick: onCardClick, ...c.props });
+          } else {
+            return React.cloneElement(c, { disabled, styles, ...c.props });
+          }
+        }
+
+        return c;
+      }),
+    [children, disabled, href, styles, onCardClick]
+  );
+
+  const hasHeading = useMemo(
+    () =>
+      React.Children.toArray(children).some(
+        (c) => React.isValidElement(c) && (c.type as any).displayName === Heading.displayName
+      ),
+    [children]
   );
 
-  const hasActions = Boolean(actions || secondaryActions);
   const disableHover = disabled || (!onClick && !href);
-  const disableEvents = disabled && !actions;
-  const onCardClick = onClick && !disabled ? onClick : undefined;
-  const onEnterKey = onClick && !disabled ? getEnterKeyHandler(onClick) : undefined;
 
   return (
-    <CardContainer
-      tabIndex={disableHover ? undefined : 0}
-      onClick={onCardClick}
-      onKeyDown={onEnterKey}
-      disableEvents={disableEvents}
-      disableHover={disableHover}
-      href={href}
-      {...htmlProps}
-    >
-      {figure}
-      <div className={styles.inner}>
-        <div className={styles.info}>
-          <div>
-            <h2 className={styles.heading}>{heading}</h2>
-            {meta}
-            {description && <p className={styles.description}>{description}</p>}
-          </div>
-          {tags}
-        </div>
-        {hasActions && (
-          <div className={styles.actionRow}>
-            {actions}
-            {secondaryActions}
-          </div>
-        )}
-      </div>
+    <CardContainer disableEvents={disabled} disableHover={disableHover} {...htmlProps}>
+      {!hasHeading && <Heading href={href} onClick={onCardClick}></Heading>}
+      {subComponents}
     </CardContainer>
   );
 };
 
-function getEnterKeyHandler(onClick: () => void) {
-  return (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter') {
-      onClick();
+interface ChildProps {
+  className?: string;
+  styles?: ReturnType<typeof getCardStyles>;
+  disabled?: boolean;
+}
+
+const Heading: FC<ChildProps & { onClick?: () => void; href?: string; 'aria-label'?: string }> = ({
+  children,
+  href,
+  onClick,
+  styles,
+  className,
+  'aria-label': ariaLabel,
+}) => {
+  return (
+    <h2 className={cx(styles?.heading, className)}>
+      {href ? (
+        <a href={href} className={styles?.linkHack} aria-label={ariaLabel}>
+          {children}
+        </a>
+      ) : onClick ? (
+        <button className={styles?.linkHack} onClick={onClick} aria-label={ariaLabel}>
+          {children}
+        </button>
+      ) : (
+        <>{children}</>
+      )}
+    </h2>
```

Comment:
I'm just not quite sure if this is the right way to produce simple and composable components. 

Ideally all `<Heading />` does is style it's children. I think then if they should be links or buttons, the consumer should pass an appropriate button or link in. I think this makes the components a bit more manageable? 

Thoughts?

Reply:
I agree with you. I guess I was trying to limit how much existing `Card` usage would need to be fixed up, but I suppose we might as well since it's a breaking change either way.

Reply:
I think you could do both.

Preferred, non-deprecated way is:

```
<Card>
  <Card.Heading>
    <Link>...</Link>
  <Card.Heading/>
</Card>
```

And then for backwards compat you can also do:

```
<Card title="..." href="..." />
```

Where the implementation is something like:

```js
const Card = ({ title, href, children }) => {
    const headingCompat = href ? <a href={href}>{title}</a> : title

    return <div>
        {title &&
            <Card.Heading>
                {headingCompat}
            </Card.Heading>
        }
        {children}
    </div>
}
```

with title and href props marked as deprecated?

Reply:
The only issue I have with this is that part of the reasoning of having `href` on the `Card` rather than the `Heading` was so the implementation detail of the `href` being on the `Heading` was elided, kind of. We want the card itself to be a link, but can't just slap the href on it because accessibility, but it can at least seem like it is by putting it on the heading together with some css tricks to cover the card. When it's something that the user passes to the heading directly, it might seem strange to then have the entire card be clickable, rather than _just_ the heading.

Reply:
Hmmm yup okay I understand...